### ISSUE TITLE
Configure multiple lexik_jwt_auth

### ISSUE THREAD

**Issue by Taluu** [lexik/LexikJWTAuthenticationBundle#707]
Kinda linked to #504, but here goes ;

It could be great to be able to set multiple configurations for the config name, by using a prototype in the config. As does doctrine for example to be able to configure multiple entity managers / connections, so we can inject the jwt authentication we need.

e.g

```yaml
lexik_jwt_authentication:
	api:
	    secret_key: 'secret_key'
	    token_ttl: 3600
	    encoder:
	        signature_algorithm: HS256

	other:
	    secret_key: 'secret_key'
	    token_ttl: 60
	    encoder:
	        signature_algorithm: HS256
```

**Comment by fl0cke** [comment]
Would be really nice to have this feature 👍

**Comment by fd6130** [comment]
Yea this multiple config for jwt is nice because we can have multiple provider in security. Hope someone could implement this feature for this bundle :(

**Comment by TonyBogdanov** [comment]
+1 for this!

In my setup I'm using JWT authentication for regular users, but I'm also using Google Cloud Pub/Sub with push endpoints, which I obviously need to firewall somehow. Google supports authenticating its requests with a JWT token generated for a service account, so all I need to do is verify that token against the service account.

The bundle works nicely to achieve this, the only problem is I can't use separate key pairs for Pub/Sub and my regular users, which is a disastrous security hole.

I'll have to explore @badrutdinovrr's custom authenticator workaround.

**Comment by chalasr** [comment]
With the new Symfony authenticator-based security system, we are probably going to move most of the config to per-firewall configuration (with a `lexik_jwt: ~` configurable authenticator). Would that fit your use cases?

**Comment by TonyBogdanov** [comment]
> With the new Symfony authenticator-based security system, we are probably going to move most of the config to per-firewall configuration (with a `lexik_jwt: ~` configurable authenticator). Would that fit your use cases?

Yes, that would actually be the best solution. I've been looking into the new approach and it really does simplify things a lot! Too bad it'll be mostly used in Symfony 6.0, which in my mind, is to be widely adopted in at least a couple of years...

**Comment by Aweptimum** [comment]
> With the new Symfony authenticator-based security system, we are probably going to move most of the config to per-firewall configuration (with a `lexik_jwt: ~` configurable authenticator). Would that fit your use cases?

Yes +1 to this, just wanted to add my support. Thank you for this awesome bundle, though!

**Comment by fd6130** [comment]
> With the new Symfony authenticator-based security system, we are probably going to move most of the config to per-firewall configuration (with a `lexik_jwt: ~` configurable authenticator). Would that fit your use cases?

How can we reuse to the other firewall if we put the config under it? Assume we want `api1` and `api2` using the same configuration (and also for the login part):

```yaml
api2:
    lexik_jwt: ~

api1:
   lexik_jwt: ~

```

**Comment by mbabker** [comment]
Something else that will help massively with supporting multiple authenticators is re-working the `AuthenticationSuccessHandler` behavior.  Right now the `onAuthenticationSuccess()` method passes only the user object into its `handleAuthenticationSuccess()` method, and the user plus the generated JWT are all that are available in the `AuthenticationSuccessEvent`.  It'd be nice if more of the security information (firewall and token) were available without having to pull data from other services which relies on order of operations not changing inside Symfony's `AuthenticatorManager` class.  I looked at this [a year ago](https://github.com/markitosgv/JWTRefreshTokenBundle/issues/242#issuecomment-872355983) in `JWTRefreshTokenBundle` and IMO having to dig around for those two bits of info makes using the `AuthenticationSuccessEvent` a bit more complicated than it needs to be (especially considering that the two events the authenticator manager dispatches around the success handler both include the token and use the firewall specific event dispatcher).

Plus, if the settings do move from the top-level bundle config to the authenticator config (or, similar to the way I handled the authenticator in the refresh token bundle, supporting "global" settings through the bundle config and overriding config values on a per-firewall basis), it becomes more important that the services used in the authentication flow become aware of the final firewall specific configuration.
